docs/src/main/tut/generate-sources-from-idl.md

```diff
@@ -0,0 +1,301 @@
+import higherkindness.mu.rpc.idlgen.IdlGenPlugin.autoImport._
+``` 
+and the setting,   
+                                                                                
+```scala  
+idlType := proto  
+sourceGenerators in Compile += (srcGen in Compile).taskValue
+```
+
+Otherwise, you can run the following sbt task (with the settings above in your `build.sbt`):
+```sbtshell
+sbt
+project foo
+srcGen   
+```
+Note that ``project foo`` it's only for multi-module projects.
+
+
+There are a couple key settings that can be configured according to various needs. In that case you must add it as the following example:
+
+```scala
+idlType := "proto",
+srcGenTargetDir := (Compile / sourceManaged).value / "mu_proto",
+srcGenSerializationType := "Protobuf",
+sourceGenerators in Compile += (srcGen in Compile).taskValue
+```
+In the example above we are using `proto` files with `Protobuf` serialization. Also, we are putting them in ``target/scala-2.12/src_managed/main/mu_proto``.
+
+Some setting options:
+
+* **`idlType`**: the type of IDL to be generated, either `proto` or `avro`.
+* **`srcGenSerializationType`**: the serialization type when generating Scala sources from the IDL definitions. `Protobuf`, `Avro` or `AvroWithSchema`(see [Schema evolution/Avro](schema-evolution/avro)) are the current supported serialization types. By default, the serialization type is 'Avro'.
+* **`srcGenSourceDirs`**: the list of directories where your IDL files are placed. By default: `Compile / resourceDirectory`, typically `src/main/resources/`.
+* **`srcGenIDLTargetDir`**: the target directory where all the IDL files specified in `srcGenSourceDirs` will be copied. Given this configuration, the plugin will automatically copy the following to this target directory:
+  * All the definitions extracted from the different `jar` or `sbt` modules, and also,
+  * All the source folders specified in the `srcGenSourceDirs` setting.
+* **`srcGenTargetDir`**: the Scala target directory, where the `srcGen` task will write the generated files in subpackages based on the namespaces declared in the IDL files. By default, `Compile / sourceManaged`, tipically  `target/scala-2.12/src_managed/main/`.
+
+The source directory must exist. Target directories will be created upon generation.
+
+There are more options:
+
+* **`genOptions`**: additional options to add to the generated `@service` annotations, after the IDL type. Currently only supports `"Gzip"`.
+* **`srcGenJarNames`**: the list of jar names or sbt modules containing the IDL definitions that will be used at compilation time by `srcGen` to generate the Scala sources. By default, this sequence is empty.
+* **`idlGenBigDecimal`**: specifies how the `decimal` types will be generated. `ScalaBigDecimalGen` produces `scala.math.BigDecimal` and `ScalaBigDecimalTaggedGen` produces `scala.math.BigDecimal` but tagged with the 'precision' and 'scale'. i.e. `scala.math.BigDecimal @@ (Nat._8, Nat._2)`. By default `ScalaBigDecimalTaggedGen`.
```

Comment:
```suggestion
* **`idlGenBigDecimal`**: specifies how the `scala.math.BigDecimal` types will be generated. `ScalaBigDecimalGen` produce the regular one and `ScalaBigDecimalTaggedGen` produces `scala.math.BigDecimal` but tagged with the 'precision' and 'scale'. i.e. `scala.math.BigDecimal @@ (Nat._8, Nat._2)`. By default is `ScalaBigDecimalTaggedGen`.
```

Reply:
Also I would add here a reference to the `http://higherkindness.io/mu/decimal-migration-guide`

Reply:
I'd keep the _specifies how the `decimal`_, since it's making reference to the decimal definition in the protocol

Reply:
Updated